src/main/java/org/zalando/nakadi/service/AuthorizationValidator.java

```diff
@@ -110,51 +109,53 @@ private void checkAuthAttributesAreValid(final Map<String, List<AuthorizationAtt
                 eventType.getName(), eventType.getAuthorization());
         try {
             final boolean authorized = authorizationService.isAuthorized(
                     null,
                     AuthorizationService.Operation.WRITE,
                     resource);
             if (!authorized) {
-                throw new ResourceAccessNotAuthorizedException(AuthorizationService.Operation.WRITE, resource);
+                throw new AccessDeniedException(null, AuthorizationService.Operation.WRITE, resource);
             }
         } catch (final PluginException ex) {
             throw new ServiceTemporarilyUnavailableException("Error while checking authorization", ex);
         }
     }
 
     public void authorizeEventTypeAdmin(final EventType eventType)
-            throws ForbiddenAccessException, ServiceTemporarilyUnavailableException {
+            throws AccessDeniedException, ServiceTemporarilyUnavailableException {
         if (eventType.getAuthorization() == null) {
             return;
         }
 
-        final Resource resource = new EventTypeResource(eventType.getName(), "event-type",
-                Collections.singletonMap(AuthorizationService.Operation.ADMIN,
-                        eventType.getAuthorization().getAdmins()));
+        final Resource resource = new EventTypeResource(eventType.getName(), eventType.getAuthorization());
         try {
             if (!authorizationService.isAuthorized(null, AuthorizationService.Operation.ADMIN, resource)) {
-                throw new ForbiddenAccessException("Editing the `EventType` is only allowed for clients that " +
-                        "satisfy the authorization `admin` requirements");
+                throw new AccessDeniedException(null, AuthorizationService.Operation.ADMIN, resource);
             }
         } catch (final PluginException e) {
             throw new ServiceTemporarilyUnavailableException("Error calling authorization plugin", e);
         }
     }
 
     public void authorizeStreamRead(final EventType eventType) throws AccessDeniedException {
-        final Resource resource = new EventTypeResource(eventType.getName(), "event-type",
-                Collections.singletonMap(AuthorizationService.Operation.READ,
-                        eventType.getAuthorization() == null ? null : eventType.getAuthorization().getReaders()));
-        final Subject subject = null;
-        if (!authorizationService.isAuthorized(subject, AuthorizationService.Operation.READ, resource)) {
-            throw new AccessDeniedException(subject, AuthorizationService.Operation.READ, resource);
+        if (eventType.getAuthorization() == null) {
+            return;
+        }
+
+        final Resource resource = new EventTypeResource(eventType.getName(), eventType.getAuthorization());
+        try {
+            if (!authorizationService.isAuthorized(null, AuthorizationService.Operation.READ, resource)) {
```

Comment:
@rcillo @v-stepanov the subject is definitely redundant with getting the principal from the security context, as we do in the plugin. However, of the 2 solutions, I think the security context may be the most generic one, as it may contain more data then what we have in the Subject interface (name and token, iirc). I can see, for example, delegation scenarios where several prinicpals are used for a single subject, which we can't represent with the Subject interface as it is.

Changing the API is not that bad, actually. It's only a plugin API, so we don't have to go through the API guild for approval.